src/infra/flow-system/sqlite/src/flow_system_event_store_sqlite.rs

```diff
@@ -0,0 +1,300 @@
+                    |mut b, (event, ReturningEventModel { event_id })| {
+                        b.push_bind(event_id);
+                        b.push_bind(fk_dataset.dataset_id.to_string());
+                        b.push_bind(fk_dataset.flow_type);
+                        b.push_bind(event.typename());
+                        b.push_bind(event.event_time());
+                        b.push_bind(serde_json::to_value(event).unwrap());
+                    },
+                );
+
+                query_builder
+            }
+            FlowKey::System(fk_system) => {
+                let mut query_builder = QueryBuilder::<Sqlite>::new(
+                    r#"
+INSERT INTO system_flow_configuration_events (event_id, system_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(
+                    events.into_iter().zip(flow_configuration_events),
+                    |mut b, (event, ReturningEventModel { event_id })| {
+                        b.push_bind(event_id);
+                        b.push_bind(fk_system.flow_type);
+                        b.push_bind(event.typename());
+                        b.push_bind(event.event_time());
+                        b.push_bind(serde_json::to_value(event).unwrap());
+                    },
+                );
+
+                query_builder
+            }
+        };
+
+        query_builder.push("RETURNING event_id");
+
+        let rows = query_builder
+            .build_query_as::<ReturningEventModel>()
+            .fetch_all(connection_mut)
+            .await
+            .int_err()?;
+
+        if let Some(last_row) = rows.last() {
+            Ok(EventID::new(last_row.event_id))
+        } else {
+            let event_id = i64::try_from(self.len().await.int_err()?).int_err()?;
+
+            Ok(EventID::new(event_id))
+        }
+    }
+
+    async fn len(&self) -> Result<usize, InternalError> {
+        let mut tr = self.transaction.lock().await;
+        let connection_mut = tr.connection_mut().await?;
+
+        let result = sqlx::query!(
+            r#"
+SELECT COUNT(event_id) as count
+FROM flow_configuration_event
+            "#,
+        )
+        .fetch_one(connection_mut)
+        .await
+        .int_err()?;
+
+        let count = usize::try_from(result.count).int_err()?;
+
+        Ok(count)
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl FlowConfigurationEventStore for FlowSystemEventStoreSqlite {
+    async fn list_all_dataset_ids(&self) -> FailableDatasetIDStream<'_> {
+        let mut tr = self.transaction.lock().await;
+
+        Box::pin(async_stream::stream! {
+            let connection_mut = tr.connection_mut().await?;
+
+            let mut query_stream = sqlx::query!(
+                r#"
+SELECT DISTINCT dataset_id
+FROM dataset_flow_configuration_events
+ORDER BY event_id
```

Comment:
As suggested in Postgres, consider narrowing the type of event.
Also, why do we need this order by?

Reply:
> As suggested in Postgres, consider narrowing the type of event.

Done

> Also, why do we need this order by?

Generally only did it for tests -- but you're right, it's not necessary at all. Removed

Reply:
Tests might want to sort loaded results in memory, that's fine.